mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -67,12 +67,127 @@ macro_rules! event {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct EndpointConfig {
+    bucket: Option<String>,
+    region: String,
+    use_fips: bool,
+    use_accelerate: bool,
+    use_dual_stack: bool,
+    endpoint: Option<Endpoint>,
+    addressing_style: AddressingStyle,
+}
+
+impl Default for EndpointConfig {
+    fn default() -> Self {
+        Self {
+            bucket: None,
+            region: "us-east-1".to_owned(),
```

Comment:
I don't think `mountpoint-s3-client` shouldn't make this assumption, even if that's what we end up doing in `mountpoint-s3`.

Didn't fully review yet, so if that sounds off - it might be! :)

Reply:
I think this probably just shouldn't implement `Default` if it includes a region field -- instead maybe there's a `EndpointConfig::default_for_region(region: &str)` method to get one.

Reply:
For the large PR, it was atomic change. I think not passing endpoint_config for each method will make this PR much shorter as you mentioned.